CRM/Activity/BAO/Activity.php

```diff
@@ -788,7 +788,14 @@ public static function getActivities($params, $getCount = FALSE) {
     // 1. we should use Activity.Getcount for fetching count only, but  in order to check that
     //    current logged in user has permission to view Case activities we are performing filtering out those activities from list (see below).
     //    This logic need to be incorporated in Activity.get definition
-    $result = civicrm_api3('Activity', 'Get', $activityParams);
+    try {
+      $result = civicrm_api3('Activity', 'Get', $activityParams);
+    }
+    catch (CiviCRM_API3_Exception $e) {
+      // CRM-20441: if the loggedin user has not enough permission to view activities
+      //  then return empty result instead of throwring error
+      return $getCount ? count($activities) : $activities;
```

Comment:
Ahh right, my 4.7.19-rc branch was outdated and it didn't had that ```try-catch``` code block.  Yup you are right.